src/Microsoft.DotNet.XHarness.TestRunners.NUnit/ResultSummary.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.DotNet.XHarness.TestRunners.Common;
+using NUnit.Engine;
+using NUnit.Framework.Interfaces;
+
+namespace Microsoft.DotNet.XHarness.TestRunners.NUnit
+{
+    /// <summary>
+    /// Helper class used to summarize the result of a test run.
+    /// </summary>
+    internal class ResultSummary : List<ITestRun>, IResultSummary
+    {
+        private readonly string _testSuite;
+        private readonly TestRunner _runner;
+        private double? _duration;
+        private long? _assertCount;
+
+        public string Name => _testSuite;
+        public string FullName => _testSuite;
+
+        public long InconclusiveTests => _runner.InconclusiveTests;
+        public long FailedTests => _runner.FilteredTests;
+        public long PassedTests => _runner.PassedTests;
+        public long SkippedTests => _runner.SkippedTests;
+        public long ExecutedTests => _runner.ExecutedTests;
+        public long TotalTests => _runner.TotalTests;
+        public long FilteredTests => _runner.FilteredTests;
+
+        public long AssertCount
+        {
+            get
+            {
+                if (_assertCount.HasValue)
+                    return _assertCount.Value;
+                // not super efficient
+                GetSummaryData();
```

Comment:
From nullability point of view, it would be better to have:
```csharp
(long, long) GetSummaryData();
```

And above something like:
```csharp
private Lazy<long> _assertCount = new Lazy<long>(() => GetSummaryData().Item1, ... any publication settings ...); 
```
or even `private readonly Lazy<(long, long)> _statistics` to not call `GetSummaryData` twice.

And then
```csharp
public long AssertCount => _statistics.AssertCount.Value; // or .Item1.Value...
```

Reply:
The issue with this is that Lazy has to call a static method. The data needs to be retrieved from the result nodes, which are an instance variable. Lazy will not help here AFAIK.